bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/console/HueCommandExtension.java

```diff
@@ -78,41 +108,120 @@ public void execute(String[] args, Console console) {
+                                    Map<String, String> lines = new TreeMap<>();
+
+                                    for (Resource resource : resources) {
+                                        MetaData metaData = resource.getMetaData();
+                                        if (Objects.nonNull(metaData)
+                                                && (metaData.getArchetype() == Archetype.BRIDGE_V2)) {
+                                            // do not list the bridge itself
+                                            continue;
+                                        }
+                                        String resourceId = resource.getId();
+                                        String idv1 = resource.getIdV1();
+                                        String thingLabel = resource.getName();
+                                        String comment = resource.getProductName();
+                                        String thingType = resourceType.name().toLowerCase();
+                                        String thingId = resourceId;
+
+                                        // special zone 'all lights'
+                                        if (resource.getType() == ResourceType.BRIDGE_HOME) {
+                                            thingLabel = clip2BridgeHandler
+                                                    .getLocalizedText(HueBindingConstants.ALL_LIGHTS_KEY);
+                                            comment = "Zone";
+                                            thingType = comment.toLowerCase();
+                                        }
+
+                                        Optional<Thing> legacyThingOptional = clip2BridgeHandler.getLegacyThing(idv1);
+                                        if (legacyThingOptional.isPresent()) {
+                                            Thing legacyThing = legacyThingOptional.get();
+                                            thingId = legacyThing.getUID().getId();
+                                            String legacyLabel = legacyThing.getLabel();
+                                            thingLabel = Objects.nonNull(legacyLabel) ? legacyLabel : thingLabel;
+                                        }
+
+                                        lines.put(thingLabel, String.format(FMT_THING, thingType, thingId, thingLabel,
+                                                resourceId, comment, idv1));
+                                    }
+                                    lines.entrySet().forEach(entry -> console.println(entry.getValue()));
+                                }
+                            }
+                            console.println("}");
+                            return;
+                    }
                 }
             }
-        } else {
-            printUsage(console);
         }
+        printUsage(console);
```

Comment:
To be moved one line before (inside the last else) to avoid a printUsage in double in certain cases.

Reply:
> one line before (inside the last else)

I think 'inside the last else' is actually TWO lines before. Or ??

Reply:
In fact, you added a return each time a command was handled. So finally, you final `printUsage(console);` is fine and you just have to delete all others (3) from this method.

Reply:
Ok. Done.